### PR TITLE
Secure Stores (biometric / password)

### PR DESCRIPTION
**Pull request by apalala-dev**
This PR adds a `SecureStore` which can be implemented either with a `BiometricStore` or a `PasswordStore`.
It is used to store and retrieve the private key.

The `BiometricStore` is backed by [biometric_storage](https://pub.dev/packages/biometric_storage). For now, it is used only on Android.
iOS will use the Secure Enclave from an other [PR](https://github.com/gabsn/starknet.dart/pull/127).

When biometry is not available (Android without fingerprint enabled, Linux, Windows, Web, old Apple devices), the `PasswordStore` will be used instead.
This store uses AES 256 in GCM mode:
- The key is a SHA 256 hash of the password
- The IV is generated randomly and has a fixed size of 16 bytes. It can be overriden but must be 16 bytes long.
The IV and the cipher text are then concatenated and base64 encoded. They are finally stored in [shared_preferences](https://pub.dev/packages/shared_preferences).
Since the IV is always 16 bytes long, we can read it from the text stored in shared_preferences and use the provided password to decrypt the encrypted private key.

This should partially address #113 .

### CODE REVIEW

File: packages/starknet_flutter/test/src/crypto/crypto_helper_test.dart [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          password: password,
+          cipherText: cipherText,
+        );
+        expect(decodedText, equals(plainText));
+      });
+    }
+  });
+
+  test("Encode with AES 256 GCM", () {
+    // TODO the password must be 16 bytes long or it will fail
```
Comment (ptisserand) [inline packages/starknet_flutter/test/src/crypto/crypto_helper_test.dart]: Password is not limited to 16 bytes, 16 bytes size is for initial vector.

File: packages/starknet_flutter/lib/src/crypto/crypto_helper.dart [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      ]),
+    );
+  }
+
+  /// Decrypt [cipherText] with AES 256 GCM using [password]. The IV will be
+  /// read from the beginning of [cipherText].
+  /// The cipherText is expected to be base64 encoded.
+  String decrypt({
+    required String password,
+    required String cipherText,
```
Comment (ptisserand) [inline packages/starknet_flutter/lib/src/crypto/crypto_helper.dart]: I think using `cipherText` in `decrypt` is a bit confusing with `encrypt` documentation:
```
  /// Encrypt [plainText] with AES 256 GCM using [password] and [iv].
  /// The returned String is the base64 encoding of the IV and the cipher text.
  /// If [iv] is not provided, a random IV will be generated using [getIV].
  ```
Maybe rename it into `encodedCipherText`

File: packages/starknet_flutter/test/src/crypto/crypto_helper_test.dart (hunk 1 above)
Comment (apalala-dev) [reply packages/starknet_flutter/test/src/crypto/crypto_helper_test.dart]: Good cath! It's an old comment, now that I've set up sha 256 it's not limited anymore 👍

File: packages/starknet/README.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+You can follow the progress of the project directly on the [kanban](https://github.com/users/gabsn/projects/1).
+
+## Contributing
+
+If you want to contribute to this project or have any suggestion, please check out our [Contributor Guide](../../CONTRIBUTING.md).
+
+## Dev Guide
+
+### Initial Setup
```
Comment (ptisserand) [inline packages/starknet/README.md:29]: Initial setup need to be run in git repository top directory where [`pyproject.toml`](../../blob/main/pyproject.toml) is located

I suggest to add these commands in top directory README and refer this file here.

File: packages/starknet/README.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+To generate docs, run:
+
+```
+dart doc .
+```
+
+For more advanced features, check out the [dartdoc package](https://pub.dev/packages/dartdoc).
+
+### Compile cairo contracts
```
Comment (ptisserand) [inline packages/starknet/README.md:78]: Cairo contracts compilation command need to be run in  top directory where [`protostar.toml`](../../blob/main/protostar.toml) is located

I suggest to add these commands in top directory README and refer this file here.

File: packages/starknet_flutter/lib/src/store/password_store.dart [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    return deleteSecret(key: privateKeyOf(id));
+  }
+
+  /// Stores the [seedPhrase] corresponding to [id] encrypted with [password].
+  /// If [iv] is not set, a random generated IV will be used.
+  Future<void> storeSeedPhrase({
+    required String password,
+    required String id,
+    required List<String> seedPhrase,
+    Uint8List? iv,
```
Comment (ptisserand) [inline packages/starknet_flutter/lib/src/store/password_store.dart:108]: `iv` parameter is missing in `storeSecret` call

**Review by ptisserand — CHANGES_REQUESTED** [review]

**Review by ptisserand — APPROVED** [review]

**Pull request merged**